# Задача 05. Надёжные вычисления
# Контекст
# Вы работаете в компании, занимающейся финансовыми вычислениями. 
# Вам необходимо разработать функцию для вычисления квадратного корня числа, 
# которая будет использоваться в анализе финансовых данных и расчёте 
# инвестиционных показателей. Вы понимаете, что передача отрицательного числа 
# или возникновение других ошибок вычисления могут привести 
# к непредсказуемым результатам.

# Задача
# Напишите функцию, которая будет:
# Вычислять и возвращать квадратный корень полученного числа.
# Обрабатывать исключения для отрицательных чисел и других возможных ошибок.

# Советы
# При помощи оператора as вы можете сохранить пойманную ошибку в переменную, 
# чтобы затем использовать её для получения дополнительной информации:
# except Exception as exc:
#     print(exc)
# Старайтесь не смешивать конкретные исключения, которые вы ожидаете, 
# со всеми другими (except Exception будет ловить все исключения, 
# которые не были пойманы предыдущими except).


import os
import math

def square_num(num):
    try:

        sqrt_num = math.sqrt(num)
        if sqrt_num == 0:
            raise Exception('корень 0 равен 0')
        
        return sqrt_num
    
    except Exception as exc:
        return exc

    except ValueError:
        return('не может быть найден так как число отрицательное')




# MAIN_CODE====================================================================

while True:
    number = float(input('введите число: '))
    if number == 0:
        
    if number:
        print('квадратный корень числа: ', square_num(number))
    else:
        break